src/Mvc/Mvc.Core/src/Infrastructure/ActionMethodExecutor.cs

```diff
@@ -48,17 +52,65 @@ public static ActionMethodExecutor GetExecutor(ObjectMethodExecutor executor)
         throw new Exception();
     }
 
+    public static ActionMethodExecutor GetFilterExecutor(EndpointFilterDelegate routeHandlerFilterDelegate) =>
+        new FilterActionMethodExecutor(routeHandlerFilterDelegate);
+
+    private sealed class FilterActionMethodExecutor : ActionMethodExecutor
+    {
+        private readonly EndpointFilterDelegate _filterDelegate;
+
+        public FilterActionMethodExecutor(EndpointFilterDelegate filterDelegate)
+        {
+            _filterDelegate = filterDelegate;
+        }
+
+        public override async ValueTask<IActionResult> Execute(
+            ActionContext actionContext,
+            IActionResultTypeMapper mapper,
+            ObjectMethodExecutor executor,
+            object controller,
+            object?[]? arguments)
+        {
+            var context = new ControllerEndpointFilterInvocationContext(actionContext, executor, mapper, controller, arguments);
+            var result = await _filterDelegate(context);
+            return ConvertToActionResult(mapper, result, executor.IsMethodAsync ? executor.AsyncResultType! : executor.MethodReturnType);
+        }
+
+        public override ValueTask<object?> Execute(ControllerEndpointFilterInvocationContext invocationContext)
+        {
+            // This is never called
+            throw new NotSupportedException();
+        }
+
+        protected override bool CanExecute(ObjectMethodExecutor executor)
+        {
+            // This is never called
+            throw new NotSupportedException();
+        }
+    }
+
     // void LogMessage(..)
     private sealed class VoidResultExecutor : ActionMethodExecutor
     {
         public override ValueTask<IActionResult> Execute(
+            ActionContext actionContext,
             IActionResultTypeMapper mapper,
             ObjectMethodExecutor executor,
             object controller,
             object?[]? arguments)
         {
             executor.Execute(controller, arguments);
-            return new ValueTask<IActionResult>(new EmptyResult());
+            return new(new EmptyResult());
```

Comment:
Is it possible to only use one single instance of `EmptyResult` instead of creating a new instance every time?